lib/flipper/typecast.rb

```diff
@@ -36,17 +36,25 @@ def self.to_float(value)
       raise ArgumentError, "#{value.inspect} cannot be converted to a float"
     end
 
-    # Internal: Convert value to a percentage.
+    # Internal: Convert value to a number.
     #
     # Returns a Integer or Float representation of the value.
     # Raises ArgumentError if conversion is not possible.
-    def self.to_percentage(value)
-      result_to_f = value.to_f
-      result_to_i = result_to_f.to_i
-      result_to_f == result_to_i ? result_to_i : result_to_f
+    def self.to_number(value)
+      case value
+      when Numeric
+        value
+      when String
+        value.include?('.') ? to_float(value) : to_integer(value)
```

Comment:
We may want to bench this against the to_f/to_i shenanigans we had.

Reply:
main:
```
Typecast.to_number 1      7.863M (± 0.3%) i/s -     40.021M in   5.090105s
Typecast.to_number 1.1
                          7.879M (± 0.3%) i/s -     39.979M in   5.074375s
Typecast.to_number '1'
                          7.004M (± 0.8%) i/s -     35.206M in   5.026764s
Typecast.to_number '1.1'
                          6.919M (± 0.8%) i/s -     35.179M in   5.085068s
Typecast.to_number nil
                          7.833M (± 0.1%) i/s -     39.786M in   5.079194s
Typecast.to_number Time.now
                          6.579M (± 0.4%) i/s -     33.543M in   5.098330s
```

this branch:
```
Typecast.to_number 1     11.578M (± 1.1%) i/s -     58.624M in   5.064082s
Typecast.to_number 1.1
                         11.503M (± 0.7%) i/s -     57.944M in   5.037778s
Typecast.to_number '1'
                          4.986M (± 0.4%) i/s -     25.126M in   5.039757s
Typecast.to_number '1.1'
                          4.805M (± 0.4%) i/s -     24.340M in   5.065364s
Typecast.to_number nil
                          7.612M (± 0.2%) i/s -     38.168M in   5.014263s
Typecast.to_number Time.now
                          5.895M (± 0.4%) i/s -     29.987M in   5.086817s
```


So with a `Numeric` it is significantly faster since it doesn't need to do anything, but definitely slower with a Sting. The problem I was trying to solve is that the current function converts `1.0` or `"1.0"` to the integer `1`. That feels wrong. I I can't think of a better way to solve it.

How often in the normal course of events are strings cast to numbers as it currently stands? Either way, I'm fine to revert this and deal with it later.

Reply:
Sweet!